source/kraken/tests/apply_disruption_test.cpp

```diff
@@ -1161,3 +1161,499 @@ BOOST_AUTO_TEST_CASE(stop_point_deletion_test) {
+            ("stop_point:20", "09:20"_t, "09:21"_t)
+            ("stop_point:30", "09:30"_t, "09:31"_t)
+            ("stop_point:40", "09:40"_t, "09:41"_t);
+    b.vj("line:A", "001100", "", true, "vj:3")
+            ("stop_point:10", "10:10"_t, "10:11"_t)
+            ("stop_point:20", "10:20"_t, "10:21"_t)
+            ("stop_point:30", "10:30"_t, "10:31"_t)
+            ("stop_point:40", "10:40"_t, "10:41"_t);
+
+    b.generate_dummy_basis();
+    b.finish();
+    b.data->pt_data->index();
+    b.data->build_raptor();
+    b.data->build_uri();
+    b.data->meta->production_date = bg::date_period(bg::date(2016,4,4), bg::days(7));
+
+    navitia::apply_disruption(b.impact(nt::RTLevel::Adapted, "line_section_on_line:A_diverted")
+                              .severity(nt::disruption::Effect::NO_SERVICE)
+                              .on_line_section("line:A", "stop_area:2", "stop_area:3", {"line:A:0"})
+                              .application_periods(btp("20160404T120000"_dt, "20160406T090000"_dt))
+                              .get_disruption(),
+                              *b.data->pt_data, *b.data->meta);
+
+    auto vjmap = b.data->pt_data->vehicle_journeys_map;
+
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->lines.size(), 1);
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->routes.size(), 1);
+    // Only 5 vj, vj:3 shouldn't be affected
+    BOOST_REQUIRE_EQUAL(b.data->pt_data->vehicle_journeys.size(), 5);
+
+    // Check the original vj
+    auto* vj = vjmap["vj:1"];
+    auto adapted_vp = vj->adapted_validity_pattern()->days;
+    auto base_vp = vj->base_validity_pattern()->days;
+    BOOST_CHECK_MESSAGE(ba::ends_with(adapted_vp.to_string(), "111001"), adapted_vp);
+    BOOST_CHECK_MESSAGE(ba::ends_with(base_vp.to_string(), "111111"), base_vp);
+
+    vj = vjmap["vj:2"];
+    adapted_vp = vj->adapted_validity_pattern()->days;
+    base_vp = vj->base_validity_pattern()->days;
+    BOOST_CHECK_MESSAGE(ba::ends_with(adapted_vp.to_string(), "101000"), adapted_vp);
+    BOOST_CHECK_MESSAGE(ba::ends_with(base_vp.to_string(), "101010"), base_vp);
+
+    vj = vjmap["vj:3"];
+    adapted_vp = vj->adapted_validity_pattern()->days;
+    base_vp = vj->base_validity_pattern()->days;
+    BOOST_CHECK_MESSAGE(ba::ends_with(adapted_vp.to_string(), "001100"), adapted_vp);
+    BOOST_CHECK_MESSAGE(ba::ends_with(base_vp.to_string(), "001100"), base_vp);
+
+    // Check the adapted vj
+    vj = vjmap["vj:1:Adapted:0:line_section_on_line:A_diverted"];
+    adapted_vp = vj->adapted_validity_pattern()->days;
+    base_vp = vj->base_validity_pattern()->days;
+    BOOST_CHECK_MESSAGE(ba::ends_with(adapted_vp.to_string(), "000110"), adapted_vp);
+    BOOST_CHECK_MESSAGE(ba::ends_with(base_vp.to_string(), "000000"), base_vp);
+    // The adapted vj should have only 2 stop_times
+    BOOST_REQUIRE_EQUAL(vj->stop_time_list.size(), 2);
+    BOOST_REQUIRE_EQUAL(vj->stop_time_list.front().stop_point->uri, "stop_point:10");
```

Comment:
not for all stop_times, but we can maybe check for the first VJ that the times are not changed (8h10 -> 8h40 in this case) no ?